pontoon/base/models/entity.py

```diff
@@ -843,24 +845,60 @@ def for_project_locale(
         if search:
             search_list = utils.get_search_phrases(search)
 
-            translation_filters = (
-                Q(translation__string__icontains_collate=(search, locale.db_collation))
-                & Q(translation__locale=locale)
-                for search in search_list
-            )
-            translation_matches = entities.filter(*translation_filters).values_list(
-                "id", flat=True
-            )
+            if search_identifiers:
+                translation_filters = (
+                    Q(
+                        translation__string__icontains_collate=(
+                            search,
+                            locale.db_collation,
+                        )
+                    )
+                    & Q(translation__locale=locale)
+                    for search in search_list
+                )
+                translation_matches = entities.filter(*translation_filters).values_list(
+                    "id", flat=True
+                )
+            else:
+                entities = entities.annotate(
+                    filtered_string=Substr("string", StrIndex("string", Value("=")) + 1)
+                )
+                translation_filters = (
+                    Q(filtered_string__icontains_collate=(search, locale.db_collation))
+                    & Q(translation__locale=locale)
+                    for search in search_list
+                )
+                translation_matches = entities.filter(*translation_filters).values_list(
```

Comment:
You can avoid duplicating `translation_matches` by simply dedenting it and removing it from the if block above.

Similarly, you also don't need to repeat `translation_filters` if you express the first `Q()` as `Q({'keyword': value})`.

Reply:
https://stackoverflow.com/a/8144600